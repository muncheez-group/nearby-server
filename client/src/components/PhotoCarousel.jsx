import React from 'react';
import { Carousel } from 'react-bootstrap';

class PhotoCarousel extends React.Component {
  constructor(props) {
    super(props);
    this.handleSelect = this.handleSelect.bind(this);

    this.state = {
      index: 0,
      direction: null
    };
  }

  handleSelect(selectedIndex, e) {
    this.setState({
      index: selectedIndex,
      direction: e.direction
    });
  }

  render() {
    const { index, direction } = this.state;
    let carousels = this.props.nearbyRestaurant.photos.map((photoID, index) => {
      if (index < 5) {
        return (
          <Carousel.Item key={index}>
            <img className="carouselImage" src={'https://s3-us-west-1.amazonaws.com/nearby-pics/' + photoID +'.jpg'} />
          </Carousel.Item>
        )
      }
    })

    return (
      <div className="restaurant-photocarousel">
        <Carousel
          activeIndex={index}
          direction={direction}
          onSelect={this.handleSelect}
        >
          {carousels}
        </Carousel>
      </div>
    )
  }
}

module.exports = PhotoCarousel;
