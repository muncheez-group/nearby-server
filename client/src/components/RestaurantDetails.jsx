import React from 'react';
import faker from 'faker';

class RestaurantDetails extends React.Component {
	constructor(props) {
		super(props);
    this.state = {
      price: []
    }
	}


	render() {
    let priceRange = [];
    for (var i = 0; i < this.props.nearbyRestaurant.price_level; i++) {
      priceRange.push('$');
    }

    let dollar = priceRange.map((dollar) => {
      return dollar;
    })

    var starsPercentage = (googleRating) => {
      var percent = googleRating / 5 * 100;
      // console.log('Percent: ', percent + '%')
      return percent + '%';
    }

		return (
			<div className="restaurant-details">
				<div className="restaurant-details-name" onClick={() => this.props.switchRestaurant(this.props.nearbyRestaurant.place_id)} >{this.props.nearbyRestaurant.name}</div>

        <div className="restaurant-details-info">
          <span>{this.props.nearbyRestaurant.types}</span><span> · </span><span>{this.props.nearbyRestaurant.neighborhood}</span><span> · </span><span>{dollar}</span>
        </div>

        <div className="restaurant-details-ratings">
          <img className="apateez-logo-small" src="https://s3-us-west-1.amazonaws.com/apateezassets/apateez-logo-small-red.jpeg" /> <span className="restaurant-details-ratings-zagat-rated">FOOD</span> <span className="restaurant-details-ratings-zagat-rated-value">{this.props.nearbyRestaurant.zagat_rating}</span> 
          <span className="pipeline">|</span>
          <img className="google-logo-small" src="https://s3-us-west-1.amazonaws.com/apateezassets/google-logo-icon.png" /><span className="restaurant-details-ratings-google-rating">{this.props.nearbyRestaurant.google_rating}</span>
          <div className="restaurant-details-ratings-stars">
            <div className="restaurant-details-ratings-stars-top" style={{width: starsPercentage(this.props.nearbyRestaurant.google_rating)}}><span>★</span><span>★</span><span>★</span><span>★</span><span>★</span></div>
            <div className="restaurant-details-ratings-stars-bottom"><span>★</span><span>★</span><span>★</span><span>★</span><span>★</span></div>
          </div>
          <span className="restaurant-details-ratings-comments">({Math.floor(Math.random()*500)})</span>
        </div>

        <div className="restaurant-details-description">
          {faker.lorem.sentence()}
        </div>        

			</div>
		)
	}
}

module.exports = RestaurantDetails;