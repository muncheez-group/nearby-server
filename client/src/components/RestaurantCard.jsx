import React from 'react';
import PhotoCarousel from './PhotoCarousel.jsx';
import RestaurantDetail from './RestaurantDetails.jsx';

class RestaurantCard extends React.Component {
	constructor(props) {
		super(props);
	}

	render() {
		return (
			<div className="restaurant-container" >
				<PhotoCarousel nearbyRestaurant={this.props.nearbyRestaurant} />
				<RestaurantDetail nearbyRestaurant={this.props.nearbyRestaurant} switchRestaurant={this.props.switchRestaurant} />
			</div>
		)
	}
}

module.exports = RestaurantCard;