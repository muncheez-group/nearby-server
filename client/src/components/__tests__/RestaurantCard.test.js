import Enzyme, { shallow, render, mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import React from 'react';
import ReactDOM from 'react-dom';
import RestaurantCard from '../RestaurantCard.jsx';
import PhotoCarousel from '../PhotoCarousel.jsx';
import RestaurantDetails from '../RestaurantDetails.jsx';

Enzyme.configure({ adapter: new Adapter() });

test('Restaurant Card shallow copy should be equal to Snapshot', () => {
    const wrapper = shallow(
        <RestaurantCard />
    );
    expect(wrapper).toMatchSnapshot();
});

const restaurant = 
    { photos: 
     [ 'CmRaAAAArs7yP4EYbcPwgoJoIBwOzTw0zTk1TV6P_vP9yaSvzduoNi2UWwivXVeGr9XgE-CiZvDEpIq_2_RontnFoLsQWi1RWnD6Llx8-_Ah5PQPDdMtWEbwovN3kldRkDxV90XsEhBy_H4_YZZMeYZC74FOcmteGhS7Faa_ZVw06Wc_Y8XeSR7JdlTbug',
       'CmRaAAAAaRvIH2tVgPOYsiYRvxBj1v-mF9C3uuloDlUfwfZFzyLAacNuU1aIRosUfU3ielQGZl-F4RKtfsLUtwbHZTw3i6LQ8jexB4iY9yV2nkmeEKljtg5fBItzqyvi2cKISnPUEhCNTLMtLLVXS7aXoxMaltpwGhSEYi4p34AJ8P8zJ01e23pBfMw0xA',
       'CmRaAAAAMRxXd0C0sx5dlxmRDJ31P9HZsCK0LtN3qTvHI84KYIe46_hXYkn24ENjlk7xmzArc47QvxJO5fKrlQPqR73RHmvfYaD5tv1pqzbu3PSriRu5CBN_mobOhcl_gxqPYxQAEhB8EGsW61qJb680opLp8QhIGhQXKapKlU36eHQRE9ddX0yUsRA3kg',
       'CmRaAAAA081C62aKoSoqrHSk3TGf3N7lvu-NR0-1Ai82PuJbmL5rdhywPYM2jAKQvL6hiDTKN3NREKasMMSEptHgD774yg2Kxu0TDu4DxFQl_5AlDOYem91-nmvI88_FCMG9cSFQEhCqFVlTa7BYhpZe7FlgCawUGhR-sTw20pnlEUhA9sIw351Uc7NUfA',
       'CmRaAAAAK2DNR3J5M6RreY0JkGtYSyJmrgdwCMxqnAXj3ZwPJHmMp0UdhhE7pw_oKUdQbil3JObB1_POTzNyjxdJ5aQpAg87F1FR9KZOf5A6Yd3v5lt4eQrkGemUXRPvNlP7pq0CEhCGUnRt0XWOChyPv-1JHWQ1GhSpJdvS2u1b7pg3ACozDBPUAiRzHw',
       'CmRaAAAAelxIapeWYaFD2OVG-WWnO5JFkgdoRRZgR7JKmC6yMhbreynt9Ud5rRxQpVKKQJJKyr8HcwwNgw0I1_QLFnoS-WTCHEZhikq_YcC1wSnaJzSeoNPQW8RJLHVhc5pQMBlSEhDnwI2eMpC1tTyegRkdBblsGhQwi6G_9Uci5exQRXLq_0OaS9NPzA',
       'CmRaAAAACWujGCbVpg_ZH3nB384vJxU1vXqTAzz0eNWK6Qy8SetrUPTZPc972pYk7SjJQwjhGJpem6ufJdYxSazKDYTxsX1Fs4WLnBuzqRfXyOr99s_1ylAdoS379htbD-BhkhqOEhDyshsi7wnMeE9MA-IL0FwXGhS58SdududKUIl8OBnU0RvQpx9scw',
       'CmRaAAAA7fWBvFQFdVgqAijZx35-fSa2oNSb_jm80lrxWWqyQRDGyGjotYn7eAG65-i9rHEHGTh4DoY5hSIdYzzIK4tMtkzH40LIblnbVpkyqT6O_6dCI09Rcm8nDrXEa19snl5TEhD-NUkT25328OuHIcoeLUn6GhTpzwqcQWIVptM19UuVVSnpSX1tTg',
       'CmRaAAAANuEuwMsU7iM2Q_gm0hQvV2vCyMPVdHvZFA30G-f4NcqeYMk1dZ1WLbJD67Xu7TDDfCRU2EYiG1bTHgN7sAGiucHDTEOz8AnXFshA_VK6J-B_iW2CETPcEh-8COfxC0AXEhC9J6oMGvdZfUGHhCCLDnMMGhRp0_IjQooHAq61lJ0fyLfRpxZecg',
       'CmRaAAAAv_BDLnr6uLEV_WW1G-xlwGnTK3Ghmyai_tQgNXbXkjc-Rn1rE5pVliramiLAphQemRa5Y5o8lPCCMB0-Fh_rhqyJNLrYgLAlWy7IB4SM8eGXnOgwjvemanZWyZtB396YEhCj2zcH7uissrhjJ39XuQekGhQ0B5cwiMUQv3cayt6ZE_xb2liWfA' ],
    nearby: 
     [ 'ChIJTel9dGCAhYARQGwrTfGZ07M',
       'ChIJcUlD0eOAhYARucGJj8eN9I8',
       'ChIJG-PkK4-AhYARxW6vm0X9kCQ',
       'ChIJWTGPjmaAhYARgXpiJ-aACYM',
       'ChIJX1E5fyZ-j4ARjujgYYPmhWA',
       'ChIJ1T-IlZiAhYAR5m-Mps4qLFk' ],
    name: 'Taverna Aventine',
    place_id: 'ChIJk2EFyIqAhYARXlKs2zO34Fg',
    google_rating: 4.2,
    zagat_rating: 1,
    neighborhood: 'Financial District',
    price_level: 2,
    types: 'restaurant',
    __v: 0 }

describe('Photo Carousel', () => {
  it('should be defined', () => {
    let wrapper = shallow(<PhotoCarousel
      nearbyRestaurant={restaurant}
    />) 
    expect(wrapper).toBeDefined();
  });
});

describe('Restaurant Detail', () => {
  it('should be defined', () => {
    let switchFunc = (id) => {
      location.href = '/restaurants/' + id;
    }
    let wrapper = shallow(<RestaurantDetails
      nearbyRestaurant={restaurant}
      switchRestaurant={switchFunc} 
    />) 
    expect(wrapper).toBeDefined();
  });
});

