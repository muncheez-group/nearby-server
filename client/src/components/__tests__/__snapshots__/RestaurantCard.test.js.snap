// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Restaurant Card shallow copy should be equal to Snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PhotoCarousel
          nearbyRestaurant={undefined}
        />,
        <RestaurantDetails
          nearbyRestaurant={undefined}
          switchRestaurant={undefined}
        />,
      ],
      "className": "restaurant-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "nearbyRestaurant": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "nearbyRestaurant": undefined,
          "switchRestaurant": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PhotoCarousel
            nearbyRestaurant={undefined}
          />,
          <RestaurantDetails
            nearbyRestaurant={undefined}
            switchRestaurant={undefined}
          />,
        ],
        "className": "restaurant-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "nearbyRestaurant": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "nearbyRestaurant": undefined,
            "switchRestaurant": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`render a label 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <RestaurantCard />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <PhotoCarousel
          nearbyRestaurant={undefined}
        />,
        <RestaurantDetails
          nearbyRestaurant={undefined}
          switchRestaurant={undefined}
        />,
      ],
      "className": "restaurant-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "nearbyRestaurant": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "nearbyRestaurant": undefined,
          "switchRestaurant": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <PhotoCarousel
            nearbyRestaurant={undefined}
          />,
          <RestaurantDetails
            nearbyRestaurant={undefined}
            switchRestaurant={undefined}
          />,
        ],
        "className": "restaurant-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "nearbyRestaurant": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "nearbyRestaurant": undefined,
            "switchRestaurant": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
